web-wallet/src/lib/components/Transactions/Transactions.svelte

```diff
@@ -1,25 +1,37 @@
 <script>
-	import { mdiContain	} from "@mdi/js";
+	import { mdiArrowLeft, mdiContain } from "@mdi/js";
 	import { fade } from "svelte/transition";
 	import { logo } from "$lib/dusk/icons";
 	import {
-		Anchor, Badge, Icon, Tooltip
+		Anchor, AnchorButton, Badge, Card, Icon, Throbber, Tooltip
 	} from "$lib/dusk/components";
 	import { createFeeFormatter, createTransferFormatter } from "$lib/dusk/currency";
-	import { settingsStore } from "$lib/stores";
 	import { calculateAdaptiveCharCount, middleEllipsis } from "$lib/dusk/string";
 	import { onMount } from "svelte";
+	import { sortByHeightDesc } from "$lib/transactions";
+	import { compose, take } from "lamb";
 
-	/** @type Transaction[] */
-	export let transactions;
+	/** @type {String} */
+	export let language;
+
+	/** @type {Number | Undefined} */
+	export let transactionLimit = undefined;
 
-	const { language } = $settingsStore;
 	const transferFormatter = createTransferFormatter(language);
 	const feeFormatter = createFeeFormatter(language);
 
+	/** @type {Promise<Transaction[]>} */
+	export let transactions;
+
 	/** @type {Number} */
 	let screenWidth = window.innerWidth;
 
+	/** @type {(transactions: Transaction[]) => Transaction[]} */
+	const getTransactionsShortlist = compose(
+		take(transactionLimit ?? 0),
```

Comment:
Rather than having a condition here, and then again checking the same condition afterwards, it's better to just have a single function.

e.g.:

```js
const getOrderedTransactions = compose(
  transactionLimit ? take(transactionLimit) : identity,
  sortByHeightDesc
);
```

Reply:
Like [so](https://github.com/dusk-network/rusk/pull/1330/commits/fa823a242390e3b78b38e500505f07c34233d05f#diff-0d7ab4782715fe40354ecd0bb87d71e2ba3a458bc4a4c7779aa2858bd85c8a34R35-R38)?

Reply:
Yes, but you can import `identity` from lamb.

Actually it's even better if you write it like this to avoid the very little overhead of `compose` and `identity` calls:

```js
const getOrderedTransactions = limit
    ? compose(take(limit), sortByHeightDesc)
    : sortByHeightDesc;
```